packages/builder/src/components/common/bindings/BindingPanel.svelte

```diff
@@ -62,15 +68,24 @@
     }
   }
 
-  // Adds a HBS helper to the expression
-  const addHelper = helper => {
-    hbsValue = addHBSBinding(hbsValue, getCaretPosition(), helper.text)
-    updateValue(hbsValue)
+  // Adds a JS/HBS helper to the expression
+  const addHelper = (helper, js) => {
+    let value
+    const pos = getCaretPosition()
+    if (js) {
+      const decoded = decodeJSBinding(jsValue)
+      value = jsValue = encodeJSBinding(
```

Comment:
The only thing here is that previously I didn't immediately update `value`, because when we call `updateValue` it first checks validity, and then dispatches the change event. That will eventually then result in the new `value` prop being passed back in here. By directly changing value it means we could be setting `value` to something that hasn't passed a validity check yet. Not saying that will definitely cause a problem, but I'd prefer to avoid updating it directly here unless it's needed.

Reply:
Makes sense, I did it this way originally as I was having problems getting the helper to be added to codemirror correctly, but I'll try and find another way!